### DISCUSSION TITLE
GUI for Lncrawler

### DISCUSSION THREAD

**Discussion by topo123** [lncrawl/lightnovel-crawler#2213] [General]
Hey, I am making a GUI for the crawler. I was just wondering if you had any ideas for how I could pull links for queries from the lncrawler when I run the command lncrawl -q "some novel title." Thanks. For context, I am using java IO and am not sure how I would do this.

**Comment by alzamer2** [comment]
hi
just asking
Lncrawler is 99% python base
why now create gui with python?

**Reply by topo123** [reply]
I am not as well versed in python as I am in Java so I decided to write it in Java. The actual method for tracking progress of the download works even though the languages are different so I do not think it is much of a problem.

**Reply by topo123** [reply]
I am ironing out some issues with the progress tracker but it works for the most part by reading the log when its updated

**Reply by thibaud200** [reply]
you can absolutly do a gui based on java/angular with python in the backend but i do not know how to make them talk to each other

**Comment by dipu-bd** [comment]
There is no need for extra GUI anymore. The web-app can be used as a GUI. I will work on updating the CLI so that the web app can be started directly in one click
